sand/controllers/search.py

```diff
@@ -1,18 +1,21 @@
 import threading
-from typing import Dict, List
+from typing import Dict, List, Union
 from flask.blueprints import Blueprint
 from sm.misc.funcs import import_func
 from sand.config import SETTINGS
 from flask import request, jsonify
 
-from sand.extension_interface.search import ISearch
+from sand.extension_interface.search import IEntitySearch, IOntologySearch
+from sand.models.search import SearchPayload
+from gena.serializer import get_dataclass_serializer
 
 search_bp = Blueprint("search", "search")
 
 GetSearchCache = threading.local()
+serializer = get_dataclass_serializer(SearchPayload)
 
 
-def get_search(name) -> ISearch:
+def get_search(name) -> Union[IEntitySearch, IOntologySearch]:
```

Comment:
Let's change the signature to be if you want to make it a single function instead of two functions:

```python
from typing import overload

@overload
def get_search(name: Literal['entities']) -> IEntitySearch: ...

@overload
def get_search(name: Literal['classes', props']) -> IOntologySearch: ...

def get_search(name: Literal['classes', 'entities', 'props']) -> Union[IEntitySearch, IOntologySearch]:
```

This will let type check work properly

Reply:
Fixed the type check using Literal types